cmd/utils.go

```diff
@@ -950,25 +950,36 @@ type timedValue struct {
 // Get will return a cached value or fetch a new one.
 // If the Update function returns an error the value is forwarded as is and not cached.
 func (t *timedValue) Get() (interface{}, error) {
-	v := t.get()
+	v := t.get(t.ttl())
 	if v != nil {
 		return v, nil
 	}
 
 	v, err := t.Update()
 	if err != nil {
+		// if update fails, return current
+		// cached value along with error.
+		//
+		// Let the caller decide if they want
+		// to use the returned value based
+		// on error.
+		v = t.get(0)
```

Comment:
t.get(0) will always return nil AFAIK, I think you want t.get(infinite), or just pass -1 and change get() to return the cached value always if ttl = -1

Reply:
oh yes @vadmeste